### DISCUSSION TITLE
Validation against a schema (feature request)

### DISCUSSION THREAD

**Discussion by DolphinDream** [nlohmann/json#3851] [Ideas]
Should the library provide a feature to validate json against a json schema ? 

To ensure json content conforms to a specific structure it would be great if you could have json data be validated against a json schema.

```
json data;  // some data structure goes here
json schema; // predefined/expected format goes here
schema.validate(data) // returns true if data conforms to schema, false otherwise
```

If this feature is not really suitable to be implemented within the library, is there a good 3rd party option that can do this kind of json schema validation (in c++, preferably working with nlohmann json) ?

**Comment by nlohmann** [comment]
You may want to try this: https://github.com/pboettch/json-schema-validator
